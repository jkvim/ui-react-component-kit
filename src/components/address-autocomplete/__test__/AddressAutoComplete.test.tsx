import * as React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { AddressAutoComplete } from '../';

describe('AddressAutoComplete', () => {

    let wrapper;

    const prefix = 'address-prefix';
    const endpoint = 'http://test.com.au';

    beforeEach(() => {
        wrapper = mount(
            <AddressAutoComplete
                prefix={prefix}
                endpoint={endpoint}
            />
        );
    });

    it('should match snapshot', () => {
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('should pass `prefix` in props', () => {
        expect(wrapper.props().prefix).toBe(prefix);
    });

    it('should pass `endpoint` in props', () => {
        expect(wrapper.props().endpoint).toBe(endpoint);
    });
});