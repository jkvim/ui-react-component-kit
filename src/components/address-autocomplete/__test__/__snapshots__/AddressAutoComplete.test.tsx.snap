// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressAutoComplete should match snapshot 1`] = `
<AddressAutoComplete
  endpoint="http://test.com.au"
  prefix="address-prefix"
>
  <AutoComplete
    dataProvider={
      AddressDataProvider {
        "endpoint": "http://test.com.au",
        "headers": Object {
          "Accept": "application/json",
        },
        "limit": 10,
      }
    }
    minSearchLength={6}
    prefix="address-prefix"
  >
    <styled.div
      onBlur={[Function]}
      role="application"
    >
      <div
        className="sc-bZQynM giHmkU"
        onBlur={[Function]}
        role="application"
      >
        <InputText
          aria-activedescendant=""
          aria-autocomplete="list"
          aria-describedby="address-prefix-autocomplete-error"
          aria-expanded={false}
          aria-invalid={false}
          aria-owns="suggestions_address-prefix"
          id="address-prefix-autocomplete"
          label=""
          onChange={[Function]}
          onKeyDown={[Function]}
          role="combobox"
          setInputRef={[Function]}
          value=""
        >
          <styled.div>
            <div
              className="sc-bxivhb fSnORn"
            >
              <styled.label
                hasError={false}
                htmlFor="address-prefix-autocomplete"
                theme={
                  Object {
                    "color": Object {
                      "color": "#333",
                      "errorColor": "#9f173f",
                    },
                  }
                }
              >
                <label
                  className="sc-bdVaJa bHemEh"
                  htmlFor="address-prefix-autocomplete"
                />
              </styled.label>
              <br />
              <styled.input
                hasError={false}
                id="address-prefix-autocomplete"
                innerRef={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder=""
                theme={
                  Object {
                    "color": Object {
                      "borderColor": "#9e9e9e",
                      "errorColor": "#9f173f",
                    },
                  }
                }
                type="text"
                value=""
              >
                <input
                  className="sc-bwzfXH rSjzs"
                  id="address-prefix-autocomplete"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  placeholder=""
                  type="text"
                  value=""
                />
              </styled.input>
              <styled.div
                theme={
                  Object {
                    "color": Object {
                      "errorColor": "red",
                    },
                  }
                }
              >
                <div
                  className="sc-EHOje RHuuV"
                />
              </styled.div>
            </div>
          </styled.div>
        </InputText>
        <div
          id="address-prefix-autocomplete-error"
        />
        <styled.div
          aria-live="polite"
        >
          <div
            aria-live="polite"
            className="sc-ifAKCX fBNUTa"
          />
        </styled.div>
      </div>
    </styled.div>
  </AutoComplete>
</AddressAutoComplete>
`;
